libraries/src/HTML/Helpers/Icons.php

```
@@ -21,6 +22,45 @@
  */
 abstract class Icons
 {
+	/**
+	 * Method to write a `<span>` element for an icon
+	 *
+	 * @param   string  $icon     The functional name for an icon.
+	 * @param   string  $srOnly   Screen Reader text if no visible text is placed
+	 * @param   array   $attribs  Attributes to be added to the wrapping element
+	 *
+	 * @return  string
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	public static function icon(string $icon = '', string $srOnly = '', array $attribs = [])
+	{
+		if ($icon === '')
+		{
+			return '';
```

Maybe overkill but do you want to throw an exception here?

---

I never implemented an exception. Can you create a PR on mine, so I can learn from it?

---

Change return to `throw new \InvalidArgumentException('error message here'); and document with `@throws`.

Again, I'm not sure if this is not overkill. I see only one instance of exception being thrown in HTML helpers. But maybe that's because we have poor sanity checks in general.

---

@SharkyKZ implemented with commit https://github.com/joomla/joomla-cms/pull/30792/commits/04c3854357a557020e9d7bf5e11a1a4ddb1ba862